StockTrader/src/main/java/com/restResource/StockTrader/repository/BuyTriggerRepository.java

```diff
@@ -0,0 +1,14 @@
+package com.restResource.StockTrader.repository;
+
+import com.restResource.StockTrader.entity.BuyTrigger;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.CrudRepository;
+
+import java.util.Optional;
+
+public interface BuyTriggerRepository extends CrudRepository<BuyTrigger, String> {
```

Comment:
Key should be a composite key of userId and stockSymbol right? See Investment class for example of how to do composite key.

Reply:
This also removes the need for the findByUserIdAndStockSymbol function, since you can just use findById

Reply:
Realized the thing with @IdClass, but you do still need to change to CrudRepository<BuyTrigger,CompositeKey>